Mathlib/Order/LiminfLimsup.lean

```diff
@@ -1517,3 +1517,98 @@ theorem liminf_finset_inf [ConditionallyCompleteLinearOrder β] [OrderTop β] {f
   limsup_finset_sup (β := βᵒᵈ) h₁ h₂
 
 end MinMax
+
+section frequently_bounded
+
+variable {R S : Type*} {F : Filter R} [LinearOrder R] [LinearOrder S]
+
+/-- For nontrivial filters in linear orders, coboundedness for `≤` implies frequently boundedness
+from below. -/
+lemma IsCobounded.frequently_ge [NeBot F] (cobdd : IsCobounded (· ≤ ·) F) :
+    ∃ l, ∃ᶠ x in F, l ≤ x := by
```

Comment:
Isn't the best result stronger than this? That is, can't you get `\forall\^f l in atTop, ...`? Or am I missing something obvious?

Reply:
I guess `\forall\^f l in atBot, ...` is possible, but in my opinion it seems to hide the usually desired existential (a value of a frequent lower bound) under a less useful quantifier. I believe in almost all use cases one would then have to use `Eventually.exists` + `Exists.choose`.

Unless you think that the generalization of the existential is valuable for some use cases, I'm tempted to keep the current form.

Reply:
Note: you *reversed* the `atTop` and `atBot` from what I intended. But it's possible what I wanted just isn't true. In that case, you can remove the new lemmas you added.

Reply:
Oh, I thought `atTop` was a misprint and should have been `atBot`. The `atTop` spelling is just false: it would say that a cobounded sequence has arbitrarily large frequent lower bounds (the constant sequence of real zeroes is a counterexample).

The combination of lemmas `IsCobounded.frequently_ge` + `IsCobounded.of_frequently_ge` says that a (nontrivial) filter is cobounded (in the `≤` direction) if and only if it has _some_ frequent lower bound. (In particular this makes it obvious that the constant sequence is cobounded and gives the counterexample to arbitrarily large frequent lower bounds.) Maybe an "iff" version would be worth including? I think I ended up wanting that in the follow-up PR.

So I remove the superficially stronger phrasing now. I indeed cannot imagine any use cases for it.

Reply:
Thanks for the explanation. Obviously I was not thinking clearly.